scripts/repository.py

```diff
@@ -0,0 +1,229 @@
+
+    for path in [cwd, *cwd.parents]:
+        if path.name == proofs:
+            return path if abspath else path_to_ancestor(cwd, path)
+        if path == root:
+            break
+    raise UserWarning(f"'{cwd}' has no ancestor named '{proofs}'")
+
+################################################################
+# Discover the roots of
+#   * the litani submodule
+#   * the starter kit submodule
+
+def load_submodules(repo=None):
+    """Load file .gitmodules describing the installed submodules"""
+
+    repo = repo or repository_root()
+    try:
+        cmd = ['git', 'config', '-f', '.gitmodules', '--list']
+        kwds = {'cwd': str(repo), 'capture_output': True, 'text': True}
+        lines = subprocess.run(cmd, **kwds, check=True).stdout.splitlines()
+    except subprocess.CalledProcessError as error:
+        logging.debug(error)
+        raise UserWarning(
+            f"Can't load submodules in repository root '{repo}'"
+        ) from error
+
+    submodules = {}
+    for line in lines:
+        line=re.sub(r'\s+', ' ', line.strip())
+
+        # Output of git config is lines of the form
+        #   submodule.ORGANIZATION/REPOSITORY.path=PATH
+        #   submodule.ORGANIZATION/REPOSITORY.url=URL
+        match = re.match(r"^submodule\.([^=]+)\.([^=]+) *= *(.+)", line)
+        if not match:
+            logging.debug("Can't parse git config output: '%s'", line)
+            continue
+
+        name, key, value = [string.strip() for string in match.groups()[0:3]]
+        submodules[name] = submodules.get(name, {})
+        submodules[name][key] = value
+    return submodules
+
+def submodule_root(url, submodules=None, repo=None, abspath=True):
+    """Look up path to root of submodule url in submodules."""
+
+    url = url.lower()
+    repo = repo or repository_root()
+    submodules = submodules or load_submodules(repo=repo)
+
+    for _, config in submodules.items():
+        config_url = config.get("url")
+        config_path = config.get("path")
+
+        if not config_url or not config_url.lower() in [url, url+".git"]:
+            continue
+        if not config_path:
+            logging.debug("Found submodule config = %s", config)
+            raise UserWarning(f"Can't find path to submodule '{url}'")
+        return Path(repo/config_path if abspath else config_path)
+
+    logging.debug("Found submodules = %s", submodules)
+    raise UserWarning(f"Can't find submodule '{url}'")
+
+def litani_root(submodules=None, repo=None, abspath=True):
+    """Root of litani submodule."""
+
+    litani = 'https://github.com/awslabs/aws-build-accumulator'
+    return submodule_root(litani, submodules, repo, abspath)
+
+def starter_kit_root(submodules=None, repo=None, abspath=True):
+    """Root of starter kit submodule."""
+
+    old_starter = 'https://github.com/awslabs/aws-templates-for-cbmc-proofs'
+    new_starter = 'https://github.com/model-checking/cbmc-starter-kit'
```

Comment:
Should we also be trying the following 2?

```
git@github.com:awslabs/aws-templates-for-cbmc-proofs.git
git@github.com:model-checking/cbmc-starter-kit.git
```

Reply:
Good catch.